### PR TITLE
Feature/needed before date

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/actions/post_test.go [hunk 1]
```diff
@@ -112,6 +115,10 @@ func (as *ActionSuite) Test_PostQuery() {
 	as.Equal(f.Posts[0].Title, resp.Post.Title)
 	as.Equal(f.Posts[0].Description.String, resp.Post.Description)
 
+	wantDate := f.Posts[0].NeededBefore.Time
+	wantDate = time.Date(wantDate.Year(), wantDate.Month(), wantDate.Day(), 0, 0, 0, 0, time.UTC)
```
Comment (Schparky) [inline application/actions/post_test.go]: Is this right? Looks the the first assignment is not used.

File: application/actions/post_test.go [hunk 2]
```diff
@@ -236,17 +245,19 @@ func (as *ActionSuite) Test_CreatePost() {
 			visibility: ALL
 		`
 	query := `mutation { post: createPost(input: {` + input + `}) { organization { id } photo { id } type title
-			description destination { description country latitude longitude }
+			neededBefore description destination { description country latitude longitude }
 			origin { description country latitude longitude }
 			size url kilograms visibility }}`
 
 	as.NoError(as.testGqlQuery(query, f.Users[0].Nickname, &postsResp))
+	//var nilTime *time.Time
```
Comment (Schparky) [inline application/actions/post_test.go]:
```suggestion
```

File: application/gqlgen/gqlgen.yml [hunk 3]
```diff
@@ -47,6 +47,8 @@ models:
     fields:
       id:
         resolver: true
+      neededBefore:
+        resolver: true
```
Comment (Schparky) [inline application/gqlgen/gqlgen.yml]: I'm just curious. Did you try it without a custom resolver?

File: application/gqlgen/post.go [hunk 4]
```diff
@@ -263,6 +273,10 @@ func convertGqlPostInputToDBPost(ctx context.Context, input postInput, currentUs
 
 	setOptionalStringField(input.Title, &post.Title)
 
+	if input.NeededBefore != nil {
+		post.NeededBefore = nulls.NewTime(*input.NeededBefore)
+	}
+
```
Comment (Schparky) [inline application/gqlgen/post.go]: Do we need a way to "erase" an existing needed-before time?

File: application/actions/post_test.go (hunk 1 above)
Comment (Baggerone) [reply application/actions/post_test.go]: It's dropping off the hours-minutes... from the fixture time.

Comment (Schparky) [reply application/actions/post_test.go]: Oh, I see it now. Sorry, read too quickly.

File: application/gqlgen/gqlgen.yml (hunk 3 above)
Comment (Baggerone) [reply application/gqlgen/gqlgen.yml]: If I don't do that, I get this ...
```
gqlgen/post.go:17:23: cannot use &postResolver literal (type *postResolver) as type PostResolver in return argument:
	*postResolver does not implement PostResolver (missing NeededBefore method)
```
Is there something else I need to add/remove, do you think?

Comment (Schparky) [reply application/gqlgen/gqlgen.yml]: Looks like its logic automatically indicates that a resolver is needed. To get that error, did you also remove your `NeededBefore` resolver?

File: application/models/models.go [hunk 5]
```diff
@@ -92,6 +92,17 @@ func GetIntFromNullsInt(in nulls.Int) *int {
 	return &output
 }
 
+// GetTimeFromNullsTime returns a pointer to a string that looks
```
Comment (Schparky) [inline application/models/models.go]:
```suggestion
// GetStringFromNullsTime returns a pointer to a string that looks
```

File: application/models/post.go [hunk 6]
```diff
@@ -363,6 +376,13 @@ func (p *Post) ValidateCreate(tx *pop.Connection) (*validate.Errors, error) {
 	//return validate.NewErrors(), nil
 }
 
+type futureDateValidator struct {
```
Comment (Schparky) [inline application/models/post.go]: Is this still needed?

File: application/models/post.go [hunk 7]
```diff
@@ -325,6 +325,16 @@ func (p *Post) SetProviderWithStatus(status PostStatus, currentUser User) {
 
 // Validate gets run every time you call a "pop.Validate*" (pop.ValidateAndSave, pop.ValidateAndCreate, pop.ValidateAndUpdate) method.
 func (p *Post) Validate(tx *pop.Connection) (*validate.Errors, error) {
+	tomorrow := time.Now()
```
Comment (Schparky) [inline application/models/post.go]: Will this work:
```
tomorrow := time.Now().Truncate(domain.DurationDay).Add(domain.DurationDay)
```

File: application/models/post.go (hunk 6 above)
Comment (Baggerone) [reply application/models/post.go]: Oh yeah. I found the built in validator after I started on that.

File: application/models/post.go (hunk 7 above)
Comment (Baggerone) [reply application/models/post.go]: That looks a lot better.

**Pull request merged**